### PR TITLE
[Merged by Bors] - feat: symmetry of the second derivative over R or C

### PR DESCRIPTION
**Pull request by sgouezel**
We expand the API around maps with symmetric second derivatives on vector spaces, to be used in #18396 for the Lie bracket of vector fields on manifolds. We also extend the fact that C^2 maps have symmetric second derivatives, already known on the reals, to a general `RCLike` field.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean [hunk 1]
```diff
@@ -47,10 +55,98 @@ rectangle are contained in `s` by convexity. The general case follows by lineari
...
+/-- Definition recording that a function has a symmetric second derivative within a set at
+a point. This is automatic in most cases of interest (open sets over real or complex vector fields,
+or general case for analytic functions), but we can express theorems of calculus using this
+as a general assumption, and then specialize to these situations. -/
+def IsSymmSndFDerivWithinAt (f : E → F) (s : Set E) (x : E) : Prop :=
+  ∀ v w, fderivWithin 𝕜 (fderivWithin 𝕜 f s) s x v w = fderivWithin 𝕜 (fderivWithin 𝕜 f s) s x w v
+
+variable (𝕜) in
+/-- Definition recording that a function has a symmetric second derivative within a set at
+a point. This is automatic in most cases of interest (open sets over real or complex vector fields,
```
Comment (jcommelin) [inline Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean:78]:
```suggestion
/-- Definition recording that a function has a symmetric second derivative within at
a point. This is automatic in most cases of interest (open sets over real or complex vector fields,
```

File: Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean [hunk 2]
```diff
@@ -47,10 +55,98 @@ rectangle are contained in `s` by convexity. The general case follows by lineari
...
+or general case for analytic functions), but we can express theorems of calculus using this
+as a general assumption, and then specialize to these situations. -/
+def IsSymmSndFDerivAt (f : E → F) (x : E) : Prop :=
```
Comment (jcommelin) [inline Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean:81]: Could you please add the lemma that projects from this predicate to `IsSymmSndFDerivWithinAt`?

File: Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean (hunk 1 above)
Comment (sgouezel) [reply Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean:78]: Not sure about this one. It's really within a set, at a point.

Comment (sgouezel) [reply Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean]: Oops, I get it now. Fixed.

File: Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean (hunk 2 above)
Comment (sgouezel) [reply Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean:81]: Done.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**